### DISCUSSION TITLE
Eclipse scrollbars prevent access to Gridbag col/row properties

### DISCUSSION THREAD

**Discussion by Tobias-Griese** [eclipse/windowbuilder#894] [Q&A] [answered]
If the window currently worked on is larger than the eclipse Tab, eclipse will display scrollbars on mouse over. These scrollbars hide the GridbagLayout columns / rows, so I cant right click a GridbagLayout column/row. Some functionality is available through the selected Gridbag cell, but for example, delete column is not .

![grafik](https://github.com/user-attachments/assets/be7451d6-63b7-4a6a-aff3-5d1e634a45ac)

Is there a way to change the layout (of eclipse or WindowManager) to decrease the eclipse-Scrollbar size, make the eclipse scrollbars displace the contents (maybe scrollbar always visible) or increase the size of the GridbagLayout columns / rows display area, so they are clickable through the scrollbar? Or is there some other way to access the functionality?

**Comment by ptziegler** [comment]
That cursor looks like Linux, right? Shouldn't the scrollbar normally be at the right/bottom of the composite? Do you have any special configuration in your IDE/in GTK that's causing this behavior?

![image](https://github.com/user-attachments/assets/84cb3e74-dd7d-417b-ab0f-754a0197c2af)

**Comment by Tobias-Griese** [comment]
I checked, because that sounded reasonable (I have default Kubuntu configuration, but maybe something changed that). Works like this in Kubuntu 24.04 fresh installation.

It's not the scrollbar of the main Design-Frame; the scrollbar only moves the small bar with the Gridbag-Columns:
![grafik](https://github.com/user-attachments/assets/134b413d-9fbd-4239-b40a-0d38ce976220)
You can see it move the colum 4 to the left, while the JFrame stays where it was.

**Reply by ptziegler** [reply, marked as answer]
Thanks for the clarification! That is indeed not how it's supposed to be. To summarize: The headers are automatically created with horizontal and vertical scrollbars enabled. Because this is done by a dependency, there is very little we can do to avoid this.

I've created https://github.com/eclipse-windowbuilder/windowbuilder/pull/903, which makes those scrollbars invisible and should effectively solve this problem. I'll try to get this merged later today, so perhaps you can try out tomorrows nightly build and see if this is still a problem on your side?

**Reply by Tobias-Griese** [reply]
Nightly build has fixed the issue. Thank you very much.
